pkg/dao/repository_configs.go

```diff
@@ -75,20 +69,9 @@ func DBErrorToApi(e error) *ce.DaoError {
 	return &ce.DaoError{Message: e.Error()}
 }
 
-func (r *repositoryConfigDaoImpl) InitializePulpClient(ctx context.Context, orgID string) error {
-	if !config.Get().Features.Snapshots.Enabled {
-		return nil
-	}
-
-	dDao := GetDomainDao(r.db)
-	domainName, err := dDao.Fetch(orgID)
-	if err != nil {
-		return err
-	}
-
-	pulpClient := pulp_client.GetPulpClientWithDomain(context.TODO(), domainName)
-	r.pulpClient = pulpClient
-	return nil
+func (r *repositoryConfigDaoImpl) WithContext(ctx context.Context) RepositoryConfigDao {
+	r.ctx = ctx
+	return r
```

Comment:
Should this return a copy of the DoaImpl instead of the same DaoImpl? 

I'm imagining two requests coming in at the same time, being run at the same time in the same process.  They are sharing a DaoImpl, so if they each call .WithContext, they could stomp over each other?  

Its very possible i'm not understanding something corretly thougH :)

Reply:
I think you're right. I'll see if I can get that to return a copy such that mocking still works in the tests.